### ISSUE TITLE
ec_deployment_template: Add data source

### ISSUE THREAD

**Issue by marclop** [elastic/terraform-provider-ec#63]
## Overview

Add a new `ec_deployment_template` datasource which returns a single deployment template.

```tf
data "ec_deployment_template" "default" {
  id     = "aws-io-optimized-v2"
  region = "us-east-1"
}
```

## Possible Implementation

Possible search or GET criteria:

* `id`: Really the only uniqueness factor.
* `name`: could be used and then an error returned when more than 1 is returned.

**Comment by marclop** [comment]
As it stands, this data source won't be useful at the moment, I'm going to remove it from from `0.1.0` and unassign it.

**Comment by marclop** [comment]
I'm marking it as `wontfix` since it's uncertain how we would benefit from it atm.

**Comment by marclop** [comment]
After #98 is merged, this won't be necessary at all since now `ec_deployment` will use the deployment templates as a base.
